### PR TITLE
Add KB for AD high memory

### PR DESCRIPTION
**Pull request by jamesesilvia**
## Description
^

## Testing
Screenshot 2024-05-29 at 11 44 17 AM

Screenshot 2024-05-29 at 11 43 29 AM

### CODE REVIEW

**Review by MichaelBaj — CHANGES_REQUESTED** [review]

File: kb/2024-05-29-I95-56484.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+**Issue ID:** I95-56484
+
+**Last Updated:** 2024-05-29
+
+**Introduced in SSR Version:** 6.2.3
+
+### Problem
+The application-director has a race condition that can occur during app-id module updates, app-id database reloads, and MIST configuration changes.
+
+It is possible that stale connections are left around to a database that will utilize excessive memory over a long duration (application-director uptime).
```
Comment (Chr1st0ph3rTurn3r) [inline kb/2024-05-29-I95-56484.md:21]:
```suggestion
It is possible that stale connections to a database will utilize excessive memory over a long duration (application-director uptime).
```

File: kb/2024-05-29-I95-56484.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+SSR
+
+### Functional Area
+System Health
+
+### Workaround
+<details>
+Juniper may provide a method to temporarily circumvent a problem; workarounds do not exist for all issues.
+</details>
+If the `application-director` is increasing memory over time, the service can be restarted on the offending device using `systemctl restart application-director` as a priviledged User.
```
Comment (Chr1st0ph3rTurn3r) [inline kb/2024-05-29-I95-56484.md:53]:
```suggestion
If the `application-director` is increasing memory over time, the service can be restarted on the offending device using `systemctl restart application-director` as a privileged User.
```

**Review by Chr1st0ph3rTurn3r — CHANGES_REQUESTED** [review]

File: kb/2024-05-29-I95-56484.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+### Resolved In
+6.2.5
+
+### Product
+SSR
+
+### Functional Area
+System Health
+
+### Workaround
```
[the remark this answers is not in the document]
Comment (jamesesilvia) [reply kb/2024-05-29-I95-56484.md:49]: Yeah. I can add some details. There are per-process memory metrics so through graphs the User should see the memory increase over time for this service.

File: kb/2024-05-29-I95-56484.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+Changes to use a new database in the app-id engine have introduced high memory utilization for the application-director.
+
+<!-- truncate -->
+
+**Issue ID:** I95-56484
+
+**Last Updated:** 2024-05-29
+
+**Introduced in SSR Version:** 6.2.3
```
Comment (agrawalkaushik) [inline kb/2024-05-29-I95-56484.md:16]: 6.2.3 or 6.2.4?

File: kb/2024-05-29-I95-56484.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+### Problem
+The application-director has a race condition that can occur during app-id module updates, app-id database reloads, and MIST configuration changes.
```
Comment (agrawalkaushik) [inline kb/2024-05-29-I95-56484.md:19]: Why just MIST config changes? The timex system we saw this on was not WAN 2.0.

File: kb/2024-05-29-I95-56484.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (agrawalkaushik) [reply kb/2024-05-29-I95-56484.md:49]: we should include the `show stats...` command down to the name of the process in this case.

File: kb/2024-05-29-I95-56484.md (hunk 5 above)
Comment (jamesesilvia) [reply kb/2024-05-29-I95-56484.md:19]: mist config changes inject apps into the AD which also causes a database reload. i can just reword to `configuration changes`

File: kb/2024-05-29-I95-56484.md (hunk 4 above)
Comment (jamesesilvia) [reply kb/2024-05-29-I95-56484.md:16]: this was caused by the sqllite changes we made in 6.2.3. one fix made it into 6.2.4 and one into 6.2.5.

File: kb/2024-05-29-I95-56484.md (hunk 5 above)
Comment (jamesesilvia) [reply kb/2024-05-29-I95-56484.md]: done

**Review by MichaelBaj — APPROVED** [review]

**Review by Chr1st0ph3rTurn3r — APPROVED** [review]

**Pull request merged**